GetIntoTeachingApiTests/Services/StoreTests.cs

```diff
@@ -532,6 +532,26 @@ public async Task GetTeachingEventBuildings_ReturnsAll()
             var result = _store.GetTeachingEventBuildings().ToList();
 
             result.Should().HaveCount(5);
+        }
+
+        [Fact]
+        public async Task SaveAsync_WillAddAndSaveGivenEntity()

```

Comment:
I would avoid the term `entity` here as that's used to reference the Dynamics entities. If you go with the suggestion above of saving the teaching event (and implicitly the building) I would maybe change this to be something like:

```
public async Task SaveAsync_WithModelAndRelatedModel_PersistsBoth()
```